### PR TITLE
Feature: Facet sorting

### PR DESCRIPTION
**Pull request by dvmartinweigl**
Closes https://github.com/datavisyn/bioinsight/issues/510, https://github.com/datavisyn/bioinsight/issues/516

### Developer Checklist (Definition of Done)

**Issue**

- [ ] All acceptance criteria from the issue are met
- [ ] Tested in latest Chrome/Firefox

**UI/UX/Vis**

- [ ] Requires UI/UX/Vis review
  - [ ] Reviewer(s) are notified (_tag assignees_)
  - [ ] Review has occurred (_link to notes_)
  - [ ] Feedback is included in this PR
  - [ ] Reviewer(s) approve of concept and design

**Code**

- [ ] Branch is up-to-date with the branch to be merged with, i.e., develop
- [ ] Code is cleaned up and formatted
- [ ] Unit tests are written (frontend/backend if applicable)
- [ ] Integration tests are written (if applicable)

**PR**

- [ ] Descriptive title for this pull request is provided (will be used for release notes later)
- [ ] Reviewer and assignees are defined
- [ ] Add type label (e.g., *bug*, *feature*) to this pull request
- [ ] Add release label (e.g., `release: minor`) to this PR following [semver](https://semver.org/)
- [ ] The PR is connected to the corresponding issue (via `Closes #...`)
- [ ] [Summary of changes](#summary-of-changes) is written

### Summary of changes
- fixed a bug in `EagerVis` causing the a wrong configuration in the scatterVis
- sort facets in scattervis and violinvis by domain or alphabetically if not given

### Additional notes for the reviewer(s)

-  
Thanks for creating this pull request 🤗

### CODE REVIEW

File: src/vis/EagerVis.tsx [hunk 1]
```diff
@@ -221,7 +221,7 @@ export function EagerVis({
 
   // eslint-disable-next-line @typescript-eslint/naming-convention
   const [_visConfig, _setVisConfig] = useUncontrolled({
-    ...(isControlled ? { value: externalConfig, onChange: setExternalConfig } : {}),
+    ...(isControlled ? { value: wrapWithDefaults(externalConfig), onChange: setExternalConfig } : {}),
```
Comment (puehringer) [inline src/vis/EagerVis.tsx:224]: The wrapWithDefaults is done right below in the useEffect as well, is there any way on how we can unify things? Or is this strictly required?

Comment (dvmartinweigl) [reply src/vis/EagerVis.tsx]: This was actually causing issues, so I again changed it to only do it in the useEffect below, but also ensured that the merge only happens once per visType

**Comment by dvmartinweigl** [comment]
> For me, the scatterplot fails as soon as it gets called in my app with the following infinite loop and console output, and causes the app to be stuck. Could you please take a look at that @dvmartinweigl ? Screenshot from 2024-08-22 15-52-40
> 
> The default sorting of the facets, which is alphabetically, works fine for me in the violin plot 👍

should be fixed now

**Comment by dvdanielamoitzi** [comment]
@dvmartinweigl 
image

**Review by dvdanielamoitzi — APPROVED** [review]
Thanks for this PR @dvmartinweigl , it works fine for me now! :+1:

**Comment by dv-usama-ansari** [comment]
Thanks for the update! I tested it locally in visyn_core stories and also in Ordino. It orders the facets alphabetically. However, I cannot test facets by domain
image
image

**Pull request merged**